# Feature reduction using PCA

In this file we will use PCA to analyse the features of the prediction dataset and possibly reduce the amount of features.

In [ ]:
import warnings
warnings.filterwarnings('ignore')

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import sklearn as sklearn
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error, mean_absolute_error, r2_score
from sklearn.preprocessing import PolynomialFeatures
from sklearn.preprocessing import StandardScaler
from sklearn.linear_model import Ridge
from sklearn.linear_model import Lasso
from sklearn.decomposition import PCA
from sklearn.model_selection import train_test_split
from matplotlib import rcParams

In [2]:
full_data = pd.read_csv("load_prediction_data.csv", parse_dates=["start_time"])
full_data.head(1)

start_time  trip_amount  start_hour  start_day  start_month  start_weekday  \
0 2017-01-01            4           0          1            1              6   

   is_weekend  mean_duration_min  max_temp  precip  
0           1                9.5      -0.6     0.0

In [24]:
X_full = full_data[["start_hour", "start_day", "start_month", "start_weekday", "is_weekend", "max_temp", "precip"]]
y_full = full_data["trip_amount"]

In [26]:
# Create a PCA model
pca = PCA()
pca.fit(X_full)
importances = pca.explained_variance_ratio_
cum = 0
for i in importances:
    print(cum + i)
    cum += i

0.4611944402300385
0.7582352621401323
0.9390233264822707
0.9835239627054879
0.9994189174536202
0.9997167128024035
1.0


Using the first 4 principle components we could capture ~ 98% of variance. However 6 features is not a lot and we do not think the added complexity of using the priciple components is worth saving on two features. 